smdebug/tensorflow/keras.py

```diff
@@ -1239,12 +1239,19 @@ def run(*args, **kwargs):
                         and hasattr(vars[0], "numpy")
                     )
                 )
-                or (not ((isinstance(grads[0], tf.Tensor)) and hasattr(grads[0], "numpy")))
             ):
                 return grads
+            # get embedding layers indexes
+            embedding_grad_indexes = set()
+            for idx, grad in enumerate(grads):
```

Comment:
Can we expect that all gradients in the list are of the same type? In which case, can we stick with the previous logic of checking the first element? If they're not all the same type, this might still miss some cases if it returns grads before checking all elements.